lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1983,27 +1987,52 @@ def _get_deployable_vhd_path(
         cached_key: Optional[bytearray] = None
         with _global_sas_vhd_copy_lock:
             blobs = container_client.list_blobs(name_starts_with=vhd_path)
+            blob_client = container_client.get_blob_client(vhd_path)
+            vhd_exists = False
             for blob in blobs:
                 if blob:
                     # check if hash key matched with original key.
                     if blob.content_settings:
                         cached_key = blob.content_settings.get("content_md5", None)
-                    if original_key == cached_key:
-                        # if it exists, return the link, not to copy again.
+                    if self._is_stuck_copying(blob_client, log):
+                        # Delete the stuck vhd.
+                        blob_client.delete_blob(delete_snapshots="include")
+                        break
```

Comment:
The blob should have only one, so it doesn't need to break. The logic will be consistent.

Reply:
Done